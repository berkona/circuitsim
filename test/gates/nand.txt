{"undoStack":[],"graph":{"0":{"nid":"0","type":"gnd","pos":{"x":280,"y":440},"text_pos":{"x":285,"y":466},"rect":{"x":270,"y":430,"width":60,"height":39},"pins":[{"pos":{"x":321,"y":457},"adj":[["1","2"]]}]},"1":{"nid":"1","type":"nmos","pos":{"x":280,"y":360},"text_pos":{"x":285,"y":365},"rect":{"x":270,"y":350,"width":60,"height":66},"pins":[{"pos":{"x":319,"y":359},"adj":[["2","2"]]},{"pos":{"x":279,"y":383},"adj":[["15","0"]]},{"pos":{"x":319,"y":407},"adj":[["0","0"]]}]},"2":{"nid":"2","type":"nmos","pos":{"x":280,"y":280},"text_pos":{"x":285,"y":285},"rect":{"x":270,"y":270,"width":60,"height":66},"pins":[{"pos":{"x":319,"y":279},"adj":[["6","0"]]},{"pos":{"x":279,"y":303},"adj":[["3","0"]]},{"pos":{"x":319,"y":327},"adj":[["1","0"]]}]},"3":{"type":"input","name":"A","pos":{"x":0,"y":160},"rect":{"x":0,"y":160,"width":25,"height":20},"pins":[{"pos":{"x":20,"y":160},"adj":[["7","1"],["2","1"]]}]},"4":{"type":"input","name":"B","pos":{"x":0,"y":320},"rect":{"x":0,"y":320,"width":25,"height":20},"pins":[{"pos":{"x":20,"y":320},"adj":[["15",0]]}]},"5":{"type":"output","name":"Y","pos":{"x":623,"y":240},"rect":{"x":623,"y":240,"width":25,"height":20},"pins":[{"pos":{"x":620,"y":240},"adj":[["6",0]]}]},"6":{"nid":"6","type":"wire","pos":{"x":320,"y":240},"rect":{"x":315,"y":235,"width":10,"height":10},"pins":[{"pos":{"x":320,"y":240},"adj":[["5","0"],["2","0"],["7","2"],["9","0"]]}]},"7":{"nid":"7","type":"pmos","pos":{"x":280,"y":160},"text_pos":{"x":285,"y":165},"rect":{"x":270,"y":150,"width":60,"height":66},"pins":[{"pos":{"x":319,"y":159},"adj":[["10","0"]]},{"pos":{"x":279,"y":182},"adj":[["3","0"]]},{"pos":{"x":319,"y":207},"adj":[["6","0"]]}]},"8":{"nid":"8","type":"pmos","pos":{"x":200,"y":100},"text_pos":{"x":205,"y":105},"rect":{"x":190,"y":90,"width":60,"height":66},"pins":[{"pos":{"x":239,"y":99},"adj":[["10","0"]]},{"pos":{"x":199,"y":122},"adj":[["15","0"]]},{"pos":{"x":239,"y":147},"adj":[["9",0]]}]},"9":{"nid":"9","type":"wire","pos":{"x":240,"y":240},"rect":{"x":235,"y":235,"width":10,"height":10},"pins":[{"pos":{"x":240,"y":240},"adj":[["8","2"],["6","0"]]}]},"10":{"nid":"10","type":"vcc","pos":{"x":240,"y":0},"text_pos":{"x":245,"y":26},"rect":{"x":230,"y":-10,"width":60,"height":39},"pins":[{"pos":{"x":281,"y":17},"adj":[["8","0"],["7","0"]]}]},"15":{"nid":"15","type":"wire","pos":{"x":200,"y":320},"rect":{"x":195,"y":315,"width":10,"height":10},"pins":[{"pos":{"x":200,"y":320},"adj":[["4","0"],["8","1"],["1","1"]]}]}},"inputPins":[{"name":"A","nid":"3"},{"name":"B","nid":"4"}],"outputPins":[{"name":"Y","nid":"5"}],"nNodes":16,"version":"1"}