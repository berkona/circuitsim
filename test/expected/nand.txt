{ "inputs": [ "A", "B" ],
  "outputs": [ "Y" ],
  "rows": 
   [ [ "1", "1", "0" ],
     [ "1", "0", "1" ],
     [ "0", "1", "1" ],
     [ "0", "0", "1" ] ] }
